llvm/test/tools/llvm-objdump/XCOFF/disassemble-abs.test

```diff
@@ -0,0 +1,37 @@
+# RUN: llvm-objdump -d %p/Inputs/abs32.o | \
```

Comment:
We can use yaml2obj to generate the object? Raw object input should always be avoided.
```
obj2yaml  abs32.o -o m.yaml
yaml2obj m.yaml -o t.o
$llvm-objdump -d t.o

t.o:	file format aixcoff-rs6000

Disassembly of section .text:

00000000 <.main>:
       0: 48 00 00 03  	bla 0
       4: 41 80 00 03  	btla	0, 0
       8: 48 00 12 36  	ba 4660
       c: 4b ff ff e2  	ba 67108832
      10: 41 80 23 4a  	bta	0, 9032
      14: 41 80 ff 02  	bta	0, 65280
```

Reply:
Thanks for the suggestion.  It looks like obj2yaml has some bugs.  The generated object files are not valid (I cannot run `ld -r t.o`).  For the 64-bit testcase, llvm-objdump fails.  I'll see what I can do.

Reply:
Is it necessary to generate an object that can pass `ld -r` for this patch? I assume we only need to correctly dump the assembly code?

Below yaml for XCOFF64 can get an object which can be disassembled by llvm-objdump.
```
--- !XCOFF
FileHeader:
  MagicNumber:     0x1F7
  NumberOfSections: 1
  CreationTime:    1701859282
  AuxiliaryHeaderSize: 0
  Flags:           0x0
Sections:
  - Name:            .text
    Address:         0x0
    Size:            0x18
    FileOffsetToData: 0x60
    FileOffsetToRelocations: 0x78
    FileOffsetToLineNumbers: 0x0
    NumberOfRelocations: 0x2
    NumberOfLineNumbers: 0x0
    Flags:           [ STYP_TEXT ]
    SectionData:     4800000341800003480012364BFFFFE24180234A4180FF02
    Relocations:
      - Address:         0x0
        Symbol:          0x1
        Info:            0x99
        Type:            0x18
      - Address:         0x6
        Symbol:          0x1
        Info:            0x8F
        Type:            0x18
Symbols:
  - Name:            .file
    Section:         N_DEBUG
    StorageClass:    C_FILE
    NumberOfAuxEntries: 0
  - Name:            .function
    Section:         N_UNDEF
    StorageClass:    C_EXT
    NumberOfAuxEntries: 1
    AuxEntries:
      - Type:                   AUX_CSECT
        StorageMappingClass:    XMC_PR
        SymbolAlignmentAndType: 0
  - Name:            ''
    Section:         .text
    StorageClass:    C_HIDEXT
    NumberOfAuxEntries: 1
    AuxEntries:
      - Type:                   AUX_CSECT
        StorageMappingClass:    XMC_PR
        SymbolAlignmentAndType: 1
  - Name:            .main
    Section:         .text
    StorageClass:    C_EXT
    NumberOfAuxEntries: 1
    AuxEntries:
      - Type:                   AUX_CSECT
        StorageMappingClass:    XMC_PR
        SymbolAlignmentAndType: 2
StringTable:
  Strings:
    - .file
    - .function
    - .text
    - .main
```

and below yaml is for XCOFF32:
```
--- !XCOFF
FileHeader:
  MagicNumber:     0x1DF
  NumberOfSections: 1
  CreationTime:    1701859282
  AuxiliaryHeaderSize: 0
  Flags:           0x0
Sections:
  - Name:            .text
    Address:         0x0
    Size:            0x18
    FileOffsetToData: 0x3C
    FileOffsetToRelocations: 0x54
    FileOffsetToLineNumbers: 0x0
    NumberOfRelocations: 0x2
    NumberOfLineNumbers: 0x0
    Flags:           [ STYP_TEXT ]
    SectionData:     4800000341800003480012364BFFFFE24180234A4180FF02
    Relocations:
      - Address:         0x0
        Symbol:          0x4
        Info:            0x99
        Type:            0x18
      - Address:         0x6
        Symbol:          0x4
        Info:            0x8F
        Type:            0x18
Symbols:
  - Name:            ''
    Section:         .text
    StorageClass:    C_HIDEXT
    NumberOfAuxEntries: 1
    AuxEntries:
      - Type:                   AUX_CSECT
        StorageMappingClass:    XMC_PR
        SymbolAlignmentAndType: 1
  - Name:            .main
    Section:         .text
    StorageClass:    C_EXT
    NumberOfAuxEntries: 1
    AuxEntries:
      - Type:                   AUX_CSECT
        StorageMappingClass:    XMC_PR
        SymbolAlignmentAndType: 2
  - Name:            .function
    Section:         N_UNDEF
    StorageClass:    C_EXT
    NumberOfAuxEntries: 1
    AuxEntries:
      - Type:                   AUX_CSECT
        StorageMappingClass:    XMC_PR
        SymbolAlignmentAndType: 0
```